### PR TITLE
Add support for sync parameter to Client.jobs.create to allow synchronous job creation

### LINKED ISSUE #27: Unable to make synchronous requests using python sjs-client

### PR DESCRIPTION
**Pull request by hakanb**
Hi there — 

This PR adds a `sync` parameter to `Client.jobs.create` to allow synchronous job creation. It simply includes `&sync=true` in the job creation URL if `sync` is set to `True` when creating the job via `Client.jobs.create`. An example is included in `README.rst`.

This should fix #27.

Thanks for your work on this project and open to any and all feedback.

### CODE REVIEW

File: sjsclient/job.py [hunk 1]
```diff
@@ -56,13 +56,14 @@ def _create_resource(self, data):
         job = self.resource_class(self, data)
         return job
 
-    def create(self, app, class_path, conf=None, ctx=None):
+    def create(self, app, class_path, conf=None, ctx=None, sync=None):
```
Comment (noorul) [inline sjsclient/job.py]: I think sync is boolean, so we should go with sync=False by default.

File: sjsclient/job.py [hunk 2]
```diff
@@ -72,6 +73,9 @@ def create(self, app, class_path, conf=None, ctx=None):
         if ctx:
             params['context'] = ctx.name
 
+        if sync is True:
```
Comment (noorul) [inline sjsclient/job.py]: If sync: would do

File: sjsclient/tests/functional/test_job.py [hunk 3]
```diff
@@ -24,12 +24,12 @@ def _create_py_app(self):
                                            app.AppType.PYTHON)
         return test_app
 
-    def _create_job(self, sjs_app, class_path, conf=None, ctx=None):
+    def _create_job(self, sjs_app, class_path, conf=None, ctx=None, sync=None):
```
Comment (noorul) [inline sjsclient/tests/functional/test_job.py]: sync=False

**Comment by noorul** [comment]
@hakanb Thank you! Will wait for travis to come back

**Comment by hakanb** [comment]
@noorul: Thank you :) 

By the way, I looked a bit in to why this PR is breaking the build — it looks like the current Travis configuration is outdated compared to the current SJS master branch (`application.conf` references `0.7.0-SNAPSHOT`, Docker installs Spark 1.6.2 instead of 2.1.0 and uses Scala 2.10 vs. 2.11), so I *think* the issue with Travis is outside the scope of this PR. 

I tried both pinning the tests to the SJS v0.7.0 tag and updating them to the SJS master branch but couldn't quite get things to work — could you give it a look please? Or let me know how I should approach this and I'll take care of it?

File: ci/Dockerfile.test [hunk 4]
```diff
@@ -11,7 +11,7 @@ COPY ci/application.conf /opt/spark-jobserver/bin/application.conf
 
 RUN sbt job-server-extras/assembly job-server-tests/package buildPython buildPyExamples
 
-RUN cp /opt/spark-jobserver/job-server-extras/target/scala-2.10/spark-job-server.jar /opt/spark-jobserver/bin/spark-job-server.jar && \
+RUN cp /opt/spark-jobserver/job-server-extras/target/scala-2.11/spark-job-server.jar /opt/spark-jobserver/bin/spark-job-server.jar && \
```
Comment (noorul) [inline ci/Dockerfile.test]: Can you revert the change in this one?

Comment (noorul) [reply ci/Dockerfile.test]: Also let add checkout 0.7.0 of spark-job-server and see how it works

after git clone may be 

RUN git checkout 8e2f459

Comment (noorul) [reply ci/Dockerfile.test]: I create branch named release/0.7.0

**Comment by hakanb** [comment]
@noorul: Would you like me to create this PR against `release/0.7.0`or merge that in to this PR?

**Comment by noorul** [comment]
@hakanb Give me sometime, I am working to figure out what is wrong with tests.

If you think you need this patch for 0.7.0, then you can raise a PR against 0.7.X. I can cut a release from that.

**Comment by hakanb** [comment]
No rush, please take your time. 👍 Thanks.

**Comment by noorul** [comment]
Can you re-base and squash commits into one?

**Comment by hakanb** [comment]
@noorul: Done 👍

**Comment by noorul** [comment]
@hakanb Thank you for the patch.
